eclass/gap-pkg.eclass

```diff
@@ -0,0 +1,408 @@
+# the old-style fake configure scripts. We also use it to find the value
+# of $GAParch, which is contained in sysinfo.gap. The "gaproot" is
+# implicitly determined by the econf call in sci-mathematics/gap. As a
+# result, calling this function requires sci-mathematics/gap at
+# build-time.
+_gap-pkg_gaproot() {
+	echo "${ESYSROOT}/usr/$(get_libdir)/gap"
+}
+
+# @FUNCTION: _gap-pkg_gaparch
+# @INTERNAL
+# @USAGE:
+# @DESCRIPTION:
+# Return the variable $GAParch from sysinfo.gap. GAP packages put their
+# executables and libraries in bin/$GAParch, and to accomplish that, we
+# sometimes need to pass $GAParch to the build system.
+_gap-pkg_gaparch() {
+	# Use a subshell so we don't pollute the environment.
+	( . $(_gap-pkg_gaproot)/sysinfo.gap && echo "${GAParch}" )
+}
+
+# @FUNCTION: gap-pkg_econf
+# @USAGE: [extra econf args]
+# @DESCRIPTION:
+# Call econf with the standard GAP package arguments that should be
+# passed to a real ./configure script:
+#
+# 1. GAP uses the ABI variable internally so we have to unset it.
+# 2. We pass the value of _gap-pkg_gaproot to --with-gaproot.
+# 3. We set both bindir and libdir to bin/$GAParch within the
+#    gap-pkg_dir. This is where Makefile.gappkg stores both
+#    types of binaries, and is therefore where GAP expects to
+#    find them.
+#
+# All arguments to gap-pkg_econf are passed through to econf.
+#
+# @EXAMPLE
+# src_configure() {
+# 	gap-pkg_econf --with-external-libsemigroups
+# }
+#
+gap-pkg_econf() {
+	econf \
+		--with-gaproot=$(_gap-pkg_gaproot) \
+		--bindir="$(gap-pkg_dir)/bin/$(_gap-pkg_gaparch)" \
+		--libdir="$(gap-pkg_dir)/bin/$(_gap-pkg_gaparch)" \
+		"${@}"
+}
+
+# @FUNCTION: gap-pkg_src_configure
+# @USAGE:
+# @DESCRIPTION:
+# Handle both "real" configure script and the fake "old-style" ones
+# still used by many GAP packages. We determine which one we're dealing
+# with by running ./configure --help; an autoconf configure script will
+# mention "PREFIX" in the output, a fake one will not.
+#
+# Real configure scripts are configured using gap-pkg_econf, fake ones
+# are executed directly with _gap-pkg_gaproot as their sole positional
+# argument.
+gap-pkg_src_configure() {
+	local _configure="${ECONF_SOURCE:-.}/configure"
+	if [[ -x ${_configure} ]] ; then
+		if ${_configure} --help | grep PREFIX &>/dev/null; then
+			# This is a real ./configure script
+			gap-pkg_econf
+		else
+			# It's an "old-style" handwritten script that does
+			# not print usage information with --help.
+			${_configure} $(_gap-pkg_gaproot) || die
+		fi
+	fi
+}
+
+# @FUNCTION: gap-pkg_src_compile
+# @USAGE:
+# @DESCRIPTION:
+# The default src_compile with the addition of V=1 to emake. The
+# Makefile.gappkg used to build most C packages defaults to a quiet
+# build without this.
+gap-pkg_src_compile() {
+	if [[ -f Makefile ]] || [[ -f GNUmakefile ]] || [[ -f makefile ]]; then
+		emake V=1 || die "emake failed"
+	fi
+}
+
+# @FUNCTION: gap-pkg_enable_tests
```

Comment:
I'm confused. It looks as if you're supposed to call this if package has tests but the test phase is run independently of whether it was called. I suppose that would lead to people missing it and effectively ending up with missing deps?

Reply:
It's just a helper function for the boilerplate associated with a package that has a test suite and has runtime deps that aren't normally build-time deps. In that case, we want to add `IUSE=test`, add `RESTRICT="!test? ( test )"`, and add all runtime deps as build-time deps when `USE=test` is set.

You could do that all yourself, of course, but it's handy to have a function that does it.

Is your issue with the phrase "enable tests?" If so, I just stole it from `distutils_enable_tests` and `elisp-enable-tests`, which serve similar purposes. (Better to be consistent than perfect.)